src/nl/rubensten/texifyidea/action/ShowEquationPreview.kt

```diff
@@ -0,0 +1,112 @@
+package nl.rubensten.texifyidea.action
+
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.SelectionModel
+import com.intellij.openapi.fileEditor.TextEditor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.wm.ToolWindow
+import com.intellij.openapi.wm.ToolWindowAnchor
+import com.intellij.openapi.wm.ToolWindowManager
+import com.intellij.psi.PsiElement
+import com.intellij.ui.content.ContentFactory
+import nl.rubensten.texifyidea.TexifyIcons
+import nl.rubensten.texifyidea.psi.LatexMathEnvironment
+import nl.rubensten.texifyidea.psi.LatexPsiUtil
+import nl.rubensten.texifyidea.window.EquationPreviewToolWindow
+import nl.rubensten.texifyidea.window.PreviewForm
+import org.jetbrains.jsonProtocol.prepareWriteRaw
+import org.picocontainer.Disposable
+
+
+/**
+ * @author Ruben Schellekens
+ */
+class ShowEquationPreview : EditorAction("Equation preview", TexifyIcons.EQUATION_PREVIEW) {
+
+    companion object {
+        @JvmStatic
+        lateinit var FORM_KEY: Key<PreviewForm>
+    }
+
+    init {
+        FORM_KEY = Key<PreviewForm>("form")
+    }
+
+
+    override fun actionPerformed(file: VirtualFile, project: Project, textEditor: TextEditor) {
+        val editor = textEditor.editor
+        val document = editor.document
+        val selection = editor.selectionModel
+        val start = selection.selectionStart
+        val end = selection.selectionEnd
+        println("Equation preview!!!")
+        var element = this.getElement(file, project, textEditor)
+        if (element != null) {
+            if (!LatexMathEnvironment::class.java.isAssignableFrom(element.javaClass)) {
+                val parent_math = LatexPsiUtil.getParentOfType(element, LatexMathEnvironment::class.java)
+//                todo: recognize math environments as well (e.g. eqnarray, etc.)
+                element = parent_math
+            }
+            if (element != null) {
```

Comment:
invert if statement with premature return. Reasoning: removes unnecessary indentation and improves readability.

```kotlin
if (element == null) return
```